### DISCUSSION TITLE
Our api endpoint returns HTTP 204 No Content which makes useQuery unhappy

### DISCUSSION THREAD

**Discussion by sndrem** [TanStack/query#6380] [Q&A]
We have an API endpoint that returns 204 in some situations, but that makes useQuery unhappy and tells us the following `Query data cannot be undefined. Please make sure to return a value other than undefined from your query function`, but I am not sure how to resolve this issue. We would like to not return a 404 from the API. 

Does anyone have any ideas on how to resolve this issue?

**Comment by TkDodo** [comment]
What your api returns and what your queryFn returns don't have to be the same thing ;)

**Reply by sndrem** [reply]
Thanks for the tip!
It didn't occur to me that I could just await the result from our api and return `null` from the queryFn if we get undefined from our api. 

My initial tests looks promising. Thank you!
